sdk/eventhub/Azure.Messaging.EventHubs/samples/Infrastructure/ISample.cs

```diff
@@ -0,0 +1,38 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.Threading.Tasks;
+
+namespace Azure.Messaging.EventHubs.Samples.Infrastructure
+{
+    /// <summary>
+    ///   Designates a class as a sample and provides a well-known means
+    ///   of executing the sample.
+    /// </summary>
+    ///
+    ///
+    public interface ISample
```

Comment:
I think it would be good to standardize on the sample architecture across all client libraries. I think your "well-known means of executing samples" only adds net value if the cost of learning about ISample can be amortized across all samples our team is producing.

Reply:
In this case, "well-known" was intended to imply "across the Event Hubs client library."   That said, I agree with the underlying point;  it would be helpful to have a standardized way of exposing samples.  I was on the fence between an attribute for discovery and an interface.   If we're moving to a something that crosses libraries, the attribute to identify samples is likely the better way to go, in my opinion.  I think there is value in having the sample clearly take parameters to indicate exactly the information it needs to execute rather than reading something implicit from the ambient context.

Personally, I'd like very much for that to be something that allows both interactive execution of the sample, so developers can launch with breakpoints and provide information without having to configure the environment.   Likewise, in order to meet the goal of keeping samples fresh, having something that allows the test infrastructure to hook samples for a smoke test is also helpful.

Reply:
In hindsight, one other thing that I'll mention is that `ISample` isn't meant to be something that is expected those using the samples need to be aware of;  they can (and should) safely ignore it.  

The intention is for that to offer type safety to SDK authors writing additional samples, ensuring that the sample signature is one that the console application shell and the Live smoke tests can pick up, instantiate, and consume.   In theory, if you trusted contributors to not make a mistake, it could be safely removed if the aforementioned "Sample" attribute existed or if we just used a convention for "anything that is in this project and starts with the word Sample is a sample" to discover them.